ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SamlDefinedIdentifiersVerifier.kt

```diff
@@ -44,69 +43,50 @@ import javax.xml.parsers.DocumentBuilderFactory
 internal class SamlDefinedIdentifiersVerifier(val node: Node) {
 
     companion object {
-        private const val ENTITY_ID_MAX_LEN = 1024
-        private const val ID_VALUE_LENGTH_LIMIT = 256
-        private const val ATTRIBUTE_NAME_FORMAT_URI =
+        internal const val ENTITY_ID_MAX_LEN = 1024
+        internal const val ID_VALUE_LENGTH_LIMIT = 256
+
+        internal const val ATTRIBUTE_NAME_FORMAT_UNSPECIFIED =
+                "urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified"
+        internal const val ATTRIBUTE_NAME_FORMAT_URI =
                 "urn:oasis:names:tc:SAML:2.0:attrname-format:uri"
-        private const val ATTRIBUTE_NAME_FORMAT_BASIC =
+        internal const val ATTRIBUTE_NAME_FORMAT_BASIC =
                 "urn:oasis:names:tc:SAML:2.0:attrname-format:basic"
 
-        private val RWEDC_URI_SET = setOf(
+        internal const val NAME_ID_FORMAT_EMAIL =
+                "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"
+        // acquired from emailregex.com
+        @Suppress("StringLiteralDuplication")
+        private val EMAIL_REGEX =
+                """
+                |((?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\
+                |x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a
+                |-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[
+                |0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*
+                |[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e
+                |-\x7f])+)\]))
+                """.trimMargin().replace("\\s".toRegex(), "")
+
+        internal val RWEDC_URI_LIST = listOf(
                 "urn:oasis:names:tc:SAML:1.0:action:rwedc-negation",
                 "urn:oasis:names:tc:SAML:1.0:action:rwedc"
         )
     }
 
     /** 8 SAML-Defined Identifiers */
     fun verify() {
-        verifyActionNamespaceIdentifiers()
```

Comment:
❓ What happened to `verifyActionNamespaceIdentifiers`?

Reply:
This was an SP test. The only thing with the "actions" is the `action` element, and the only thing with an `action` element is the `authzDecisionQuery` element, which is sent by the SP.

I'm not sure what the response to that query looks like; but it is probably just a response with assertions, and we aren't testing that protocol in the first place.